src/doc/man/cargo-metadata.md

```diff
@@ -50,6 +53,8 @@ The output has the following format:
                  Example: "registry+https://github.com/rust-lang/crates.io-index"
                - "git+URL" for git-based dependencies.
                  Example: "git+https://github.com/rust-lang/cargo?rev=5e85ba14aaa20f8133863373404cb0af69eeef2c#5e85ba14aaa20f8133863373404cb0af69eeef2c"
+               Consumers should generally see it as an opaque string. There is
+               no guarantee of the format across different versions of Cargo.
             */
             "source": null,
```

Comment:
Interesting that `cargo_metadata` [doesn't make mention](https://docs.rs/cargo_metadata/latest/cargo_metadata/struct.Source.html) of this being opaque.  Might be worthwhile to reach out to see if there is any concern from the maintainers abotu this

Reply:
Filed one: https://github.com/oli-obk/cargo_metadata/issues/237

Reply:
From the author of `cargo_metadata`:

> That's just an oversight. We should document it as opaque, too

And the PR was merged yesterday — <https://github.com/oli-obk/cargo_metadata/pull/238>.